libdino/src/service/muc_manager.vala

```diff
@@ -28,14 +28,14 @@ public class MucManager : StreamInteractionModule, Object {
         stream_interactor.get_module(MessageProcessor.IDENTITY).pre_message_received.connect(on_pre_message_received);
     }
 
-    public void join(Account account, Jid jid, string? nick, string? password) {
+    public void join(Account account, Jid jid, string? nick, string? password, string? history_since) {
```

Comment:
Could you please not make history_since an argument here, but instead determine the value in this function? The way it is now, only history for autojoined MUCs but not for manually joined MUCs would be properly handled.